storage/innobase/include/trx0purge.h

```diff
@@ -244,15 +242,23 @@ class purge_sys_t
 					record */
 	uint16_t	hdr_offset;	/*!< Header byte offset on the page */
 
+  /** Binary min-heap of indexes in trx_sys.rseg_array, ordered on
+  rseg_t::last_trx_no(). It is protected by the pq_mutex */
+  purge_queue_t purge_queue;
+
+  /** Mutex protecting purge_queue */
+  mysql_mutex_t pq_mutex;
 
-	TrxUndoRsegsIterator
-			rseg_iter;	/*!< Iterator to get the next rseg
-					to process */
 public:
-	purge_pq_t	purge_queue;	/*!< Binary min-heap, ordered on
-					TrxUndoRsegs::trx_no. It is protected
-					by the pq_mutex */
-	mysql_mutex_t	pq_mutex;	/*!< Mutex protecting purge_queue */
+  /** Push to purge queue without acquiring pq_mutex.
+  @param rseg rseg to push */
+  void enqueue(trx_rseg_t &rseg);
+
+  /** Acquare purge_queue_mutex */
+  void lock_purge_queue_mutex() { mysql_mutex_lock(&pq_mutex); }
+
+  /** Release purge queue mutex */
+  void unlock_purge_queue_mutex() { mysql_mutex_unlock(&pq_mutex); }
```

Comment:
The `_mutex` in these names is a bit redundant, and the original InnoDB naming pattern was `noun_verb`, like `purge_queue_lock()` or maybe `queue_lock()` as the `purge` is already part of the name of the singleton object: `purge_sys.queue_lock()`.